EventFilter/L1TXRawToDigi/plugins/L1TTwinMuxRawToDigi.cc

```diff
@@ -369,347 +380,542 @@ void L1TTwinMuxRawToDigi::processFed( int twinMuxFed,
       long dataWordSub = (*DTTM7iterator);
       int selector = ( dataWordSub >> 60 ) & 0xF; // positions 60 -> 63
 
-      if ( selector == 0x4 ) { //TSC word
-
-        bxID = ( dataWordSub >> 48 ) & 0xFFF; // positions 48 -> 60
-        bc0  = ( dataWordSub >> 22 ) & 0x1; // positions 22 -> 23
-        bxNr = normBx(bxID, bxCounter); /// bx normalized to the bxcounter
+	if (firmware_rev >= fw_rev_with_zerosupression ){ // zerosuppressed data unpacking 
+		if ( selector == 0xC ) { /// BX info + theta info word
+			
+			bc0  = ( dataWordSub >> 59 ) & 0x1; // position 59
+			/*** NOT UNPACKED  
+			int L1A = ( dataWordSub >> 58) & 0x1;    // position 58
+			***/
+			int bxOffset = ( dataWordSub >> 48 ) & 0x3F; // positions 48 -> 53
+			/// Two's complement conversion
+			/// Important: here value 0x20 has been treated as -32 when in fact it is an error code
+			/// denoting incorrect readout window configuration, but the case should really not happen
+			bxNr = bxOffset < 32 ? bxOffset : bxOffset - 64;
+
+				
+
+			//eta info
+		   
+			int posBTI[7], qualBTI[7];
+			
+			int mb3_eta    = ( dataWordSub & 0xFF );        // positions  0 -> 7
+			int mb3_eta_HQ = ( dataWordSub >> 8  ) & 0xFF;  // positions  8 -> 15
+			int mb2_eta    = ( dataWordSub >> 16 ) & 0xFF;  // positions 16 -> 23
+			int mb2_eta_HQ = ( dataWordSub >> 24 ) & 0xFF;  // positions 24 -> 31
+			int mb1_eta    = ( dataWordSub >> 32 ) & 0xFF;  // positions 32 -> 39
+			int mb1_eta_HQ = ( dataWordSub >> 40 ) & 0xFF;  // positions 40 -> 47
+			
+			//MB1
+			for (int i = 0 ; i<=5; i++){
```

Comment:
hi @carstenheidemann - by putting mb*_eta and mb*_eta_HQ into arrays you can remove this code duplication (replacing it by a loop)